conductor/lib/maya_utils.py

```diff
@@ -35,9 +35,93 @@ def get_short_name(node):
 
 
 def get_maya_version():
+    '''
+    Return the version string of the currently running maya session. 
+    e.g. "2015"
+    '''
     return cmds.about(version=True)
 
 
+def get_plugin_versions():
+    '''
+    Query maya for all of it's plugins and their  versions. 
+    
+    return a dictionary such as:
+        {'AbcExport': '1.0',
+         'AbcImport': '1.0',
+         'BifrostMain': '1.0',
+         'HoldOut': '1.0',
+         'MayaMuscle': '2.00 (Build: 004)',
+         'Mayatomr': '2015.0 - 3.12.1.18 ',
+         'Substance': '1.11143',
+         'autoLoader': '1.0',
+         'bifrostshellnode': '2015',
+         'bifrostvisplugin': '3.0',
+         'cgfxShader': 'cgfxShader 4.5 for Maya 2015.0 (Apr  9 2015)',
+         'fbxmaya': '2015.1',
+         'glmCrowd': '4.1.3[21f4d33]-2015/08/25',
+         'gpuCache': '1.0',
+         'iDeform': '1.0',
+         'ik2Bsolver': '2.5',
+         'ikSpringSolver': '1.0',
+         'matrixNodes': '1.0',
+         'mayaCharacterization': '5',
+         'mayaHIK': '1.0_HIK_2014.2',
+         'modelingToolkit': 'Unknown',
+         'modelingToolkitStd': '0.0.0.0',
+         'quatNodes': '1.0',
+         'relax_node': 'Unknown',
+         'retargeterNodes': '1.0',
+         'rotateHelper': '1.0',
+         'sceneAssembly': '1.0',
+         'shaderFXPlugin': '1.0',
+         'skinningDecomposition': '1.0',
+         'spReticleLoc': '2.0',
+         'tiffFloatReader': '8.0',
+         'vrayformaya': '3.00.01',
+         'xgenMR': '1.0',
+         'xgenToolkit': '1.0'}
+    
+    '''
+    plugin_versions = {}
+    for plugin in cmds.pluginInfo(q=True, listPlugins=True):
+        plugin_versions[str(plugin)] = str(cmds.pluginInfo(plugin, version=True, q=True))
+    return plugin_versions
+
+
+
+
+def derive_docker_image(version, plugin_versions):
```

Comment:
This shit needs to be more sane (i.e. use a formalized version naming/matching convention/mechanism).  But good enough for this week.